src/coreclr/vm/typestring.cpp

```diff
@@ -387,10 +366,11 @@ HRESULT TypeNameBuilder::AddArray(DWORD rank)
         else             // allocation OK, do it the fast way
```

Comment:
This whole thing about trying to be robust in the presence of allocation failures is way overengineered. You can change this to just stackalloc the buffer while you are on it.